src/main/java/jenkins/plugins/office365connector/Office365ConnectorWebhookNotifier.java

```diff
@@ -479,38 +477,33 @@ private String evaluateMacro(String template) {
 
     private void pullRequestActionable(List<PotentialAction> paList, List<Fact> factsList) {
         Job job = run.getParent();
-        ItemGroup parent = job.getParent();
-        if (parent instanceof MultiBranchProject) {
-            BranchProjectFactory projectFactory = ((MultiBranchProject) parent).getProjectFactory();
-            if (projectFactory.isProject(job)) {
-                Branch branch = projectFactory.getBranch(job);
-                SCMHead head = branch.getHead();
-
-                if (head instanceof ChangeRequestSCMHead) {
-                    String pronoun = StringUtils.defaultIfBlank(head.getPronoun(), "Change Request");
-                    String viewName = String.format("View %s", pronoun);
-                    String titleName = String.format("%s Title", pronoun);
-                    String authorName = String.format("%s Author", pronoun);
-
-                    ObjectMetadataAction oma = branch.getAction(ObjectMetadataAction.class);
-                    if (oma != null) {
-                        String urlString = oma.getObjectUrl();
-                        PotentialAction viewPRPotentialAction = new PotentialAction(viewName, urlString);
-                        paList.add(viewPRPotentialAction);
-                        factsList.add(new Fact(titleName, oma.getObjectDisplayName()));
-                    }
-                    ContributorMetadataAction cma = branch.getAction(ContributorMetadataAction.class);
-                    if (cma != null) {
-                        String contributor = cma.getContributor();
-                        String contributorDisplayName = cma.getContributorDisplayName();
-                        String author = StringUtils.defaultIfBlank(cma.getContributor(), cma.getContributorDisplayName());
-                        if (StringUtils.isNotBlank(contributor) && StringUtils.isNotBlank(contributorDisplayName))
-                            author = String.format("%s (%s)", cma.getContributor(), cma.getContributorDisplayName());
-
-                        if (StringUtils.isNotBlank(author))
-                            factsList.add(new Fact(authorName, author));
-                    }
-                }
+        SCMHead head = SCMHead.HeadByItem.findHead(job);
+        if (head instanceof ChangeRequestSCMHead) {
```

Comment:
@casz 
I'm trying to write unit test for this code but I can't trigger the `run` so this condition is true. I was trying to use [github plugin](https://wiki.jenkins.io/display/JENKINS/Github+Plugin) and push/pull change that launches webhook to Jenkins but this condition is still false.
Can you tell me how to start SCM change so this code is executed? As right now this code seems to be dead :(

Reply:
You have to use the github/bitbucket/gitlab/gitea branch source plugin

Reply:
This code is VERY much alive!

Reply:
See this lovely screenshot ![](https://user-images.githubusercontent.com/1661688/31313997-5d904aa2-abf4-11e7-8129-638bba8d96b0.png)

Reply:
@casz 
Screenshot is cool so I believe the code is working :)
However I was not able to meet condition that adds those facts. I was trying to make it with github plugin, the build starts on push event (fired by github.com) but above condition was not met so the facts were not added.

Reply:
github plugin is not the same as [github branch source plugin](https://wiki.jenkins.io/display/JENKINS/GitHub+Branch+Source+Plugin)

Reply:
Very helpful - I was able to generate JSON with notification so now I can fill the test coverage gap :)